### PR TITLE
Add desktop entry m4 template

### PR DESCRIPTION
**Pull request by Maxr1998**
Instead of making packagers ship this themselves, this should be a predefined template in the upstream repo.

### CODE REVIEW

File: assets/finamp.desktop.m4 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Type=Application
+Name=Finamp
+GenericName=Music Player
+Icon=finamp
+Exec=__INSTALL_PATH__/finamp
+Terminal=false
+Categories=AudioVideo;Audio;Player;Music;
+Comment=A Jellyfin music client for mobile and desktop
```
Comment (Chaphasilor) [inline assets/finamp.desktop.m4:9]: please use the description from pubspec.yaml:

> An open source Jellyfin music player

File: assets/finamp.desktop.m4 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Type=Application
+Name=Finamp
+GenericName=Music Player
+Icon=finamp
```
Comment (Chaphasilor) [inline assets/finamp.desktop.m4:5]: does that works already, or do maintainers need to adapt that to their package?

File: assets/finamp.desktop.m4 [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+Exec=__INSTALL_PATH__/finamp
```
Comment (Chaphasilor) [inline assets/finamp.desktop.m4:6]: is this a variable that needs to be changed? if so, a comment (if possible) would be nice.

**Comment by Maxr1998** [comment]
(this will be used by the AUR package, which currently ships this file itself)

**Review by Chaphasilor — COMMENTED** [review]
Thanks!

Maybe you could also throw in a section into CONTRIBUTING.md to mention that this template exists?

File: assets/finamp.desktop.m4 (hunk 2 above)
Comment (Maxr1998) [reply assets/finamp.desktop.m4:5]: As long as the icons are also installed according to the [XDG icon theme spec][spec], yes. 

[spec]: https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html

File: assets/finamp.desktop.m4 (hunk 3 above)
Comment (Maxr1998) [reply assets/finamp.desktop.m4:6]: It does, it's meant to be templated with m4. I'll add a note to CONTRIBUTING.md, that should be ok.

Comment (Maxr1998) [reply assets/finamp.desktop.m4:7]: Note added, I think this should be enough for people packaging the app for their distribution.

**Comment by Chaphasilor** [comment]
perfect, thanks!

**Pull request merged**